src/backend/distributed/operations/shard_split.c

```diff
@@ -955,6 +939,46 @@ InsertSplitChildrenShardMetadata(List *shardGroupSplitIntervalListList,
 }
 
 
+/*
+ * CreatePartitioningHierarchy creates the partitioning
+ * hierarchy between the shardList, if any.
+ */
+static void
+CreatePartitioningHierarchy(List *shardGroupSplitIntervalListList,
+							List *workersForPlacementList)
+{
+	/* Create partition heirarchy between shards */
+	List *shardIntervalList = NIL;
+
+	/*
+	 * Iterate over all the shards in the shard group.
+	 */
+	foreach_ptr(shardIntervalList, shardGroupSplitIntervalListList)
+	{
+		ShardInterval *shardInterval = NULL;
+		WorkerNode *workerPlacementNode = NULL;
+
+		/*
+		 * Iterate on split shards list for a given shard and create constraints.
+		 */
+		forboth_ptr(shardInterval, shardIntervalList, workerPlacementNode,
+					workersForPlacementList)
+		{
+			if (PartitionTable(shardInterval->relationId))
+			{
+				char *attachPartitionCommand =
+					GenerateAttachShardPartitionCommand(shardInterval);
+
+				SendCommandToWorker(
```

Comment:
should this be part of the transaction or done in a batch outside of the transaction? (not sure)

Reply:
Don't see any advantages of doing it outside transaction. With foreign key to reference tables, I was also seeing some weird deadlocks if I went with outside transaction so decided to keep it in batch (Probably same reason here too:  https://github.com/citusdata/citus/blob/6a32061c08ae202a0d4d5fafbdaf9604697dd51b/src/backend/distributed/operations/isolate_shards.c#L349)

Reply:
Helpful comment. Added.